src/Core/InstanceConfigTrait.php

```diff
@@ -78,29 +78,100 @@
      * $this->config(['one' => 'value', 'another' => 'value']);
      * ```
      *
-     * @param string|array|null $key The key to get/set, or a complete array of configs.
+     * @param string|array $key The key to set, or a complete array of configs.
      * @param mixed|null $value The value to set.
      * @param bool $merge Whether to recursively merge or overwrite existing config, defaults to true.
-     * @return mixed Config value being read, or the object itself on write operations.
+     * @return $this
      * @throws \Cake\Core\Exception\Exception When trying to set a key that is invalid.
      */
-    public function config($key = null, $value = null, $merge = true)
+    public function setConfig($key, $value = null, $merge = true)
     {
         if (!$this->_configInitialized) {
             $this->_config = $this->_defaultConfig;
             $this->_configInitialized = true;
         }
 
-        if (is_array($key) || func_num_args() >= 2) {
-            $this->_configWrite($key, $value, $merge);
+        $this->_configWrite($key, $value, $merge);
+
+        return $this;
+    }
 
-            return $this;
+    /**
+     * Sets the config, either by key or passed as array.
+     *
+     * ### Usage
+     *
+     * Setting a specific value:
+     *
+     * ```
+     * $this->setConfig('key', $value);
+     * ```
+     *
+     * Setting a nested value:
+     *
+     * ```
+     * $this->setConfig('some.nested.key', $value);
+     * ```
+     *
+     * Updating multiple config settings at the same time:
+     *
+     * ```
+     * $this->setConfig(['one' => 'value', 'another' => 'value']);
+     * ```
+     *
+     * @param string|null $key The key to get or null for the whole config.
+     * @return mixed Config value being read.
+     */
+    public function getConfig($key)
```

Comment:
Since the docblock says "null for the whole config" shouldn't the default value for `$key` be `null` ?